src/join/composer.jl

```diff
@@ -96,8 +96,16 @@ function compose_inner_table(joiner::DataFrameJoiner,
                              left_rename::Union{Function, AbstractString, Symbol},
                              right_rename::Union{Function, AbstractString, Symbol})
     left_ixs, right_ixs = find_inner_rows(joiner)
-    dfl = joiner.dfl[left_ixs, :]
-    dfr_noon = joiner.dfr[right_ixs, Not(joiner.right_on)]
+
+    if VERSION >= v"1.4" && Threads.nthreads() > 1 && length(left_ixs) >= 1_000_000
+        dfl_task = Threads.@spawn joiner.dfl[left_ixs, :]
```

Comment:
since `getindex` already uses threading this is only to ensure we are fast in case left and right table have very different processing times.